packages/primitives/popover/src/popover-root.directive.ts

```diff
@@ -0,0 +1,154 @@
+    standalone: true,
+    providers: [
+        {
+            provide: RdxPopoverRootToken,
+            useExisting: forwardRef(() => RdxPopoverRootDirective)
+        }
+    ],
+    exportAs: 'rdxPopoverRoot'
+})
+export class RdxPopoverRootDirective implements OnInit {
+    /**
+     * The open state of the popover when it is initially rendered. Use when you do not need to control its open state.
+     */
+    readonly defaultOpen = input<boolean>(false);
+    /**
+     * The controlled open state of the popover. Must be used in conjunction with onOpenChange.
+     */
+    readonly open = input<boolean | undefined>();
+
+    /**
+     * Event handler called when the open state of the popover changes.
+     */
+    readonly onOpenChange = output<boolean>();
+
+    /** @ignore */
+    readonly popoverContentDirective = contentChild.required(RdxPopoverContentDirective);
+    /** @ignore */
+    readonly popoverTriggerDirective = contentChild.required(RdxPopoverTriggerDirective);
+    /** @ignore */
+    readonly popoverArrowDirective = contentChild(RdxPopoverArrowToken);
+
+    /** @ignore */
+    readonly viewContainerRef = inject(ViewContainerRef);
+    /** @ignore */
+    private readonly document = inject(DOCUMENT);
+
+    /** @ignore */
+    readonly isOpen = signal<boolean>(this.defaultOpen());
+    /** @ignore */
+    readonly state = computed<RdxPopoverState>(() => {
+        const currentIsOpen = this.isOpen();
+        if (currentIsOpen) {
+            return 'open';
+        }
+        return 'closed';
+    });
+
+    /** @ignore */
+    private isControlledExternally = computed(() => signal(this.open() !== void 0));
+
+    /** @ignore */
+    ngOnInit(): void {
+        if (this.defaultOpen()) {
+            this.handleOpen();
+        }
+    }
+
+    /** @ignore */
+    controlledExternally() {
+        return this.isControlledExternally().asReadonly();
+    }
+
+    /** @ignore */
+    handleOpen(): void {
+        if (this.isControlledExternally()()) {
+            return;
+        }
+        this.setOpen(true);
+    }
+
+    /** @ignore */
+    handleClose(): void {
+        if (this.isControlledExternally()()) {
+            return;
+        }
+        this.setOpen(false);
+    }
+
+    /** @ignore */
+    handleToggle(): void {
+        if (this.isControlledExternally()()) {
+            return;
+        }
+        this.isOpen() ? this.handleClose() : this.handleOpen();
+    }
+
+    /** @ignore */
+    private setOpen(open = false): void {
+        if (open) {
+            this.document.dispatchEvent(new CustomEvent('popover.open'));
+        }
+
+        this.isOpen.set(open);
+        this.onOpenChange.emit(open);
+    }
+
+    /** @ignore */
+    private show(): void {
+        this.popoverContentDirective().show();
+    }
+
+    /** @ignore */
+    private hide(): void {
+        this.popoverContentDirective().hide();
+    }
+
+    /** @ignore */
+    private readonly onIsOpenChangeEffect = effect(() => {
```

Comment:
I don't see this method being used.

Reply:
this is a leftover from the `tooltip` - I thought it is a patter here to assign `EffectRef` to a property instead of calling it directly or via methid in a constructor.

Will change :)

Reply:
I see this get merged so I will fix it within another PR